frontend/views/pages/Contributions.vue

```diff
@@ -129,25 +197,66 @@ export default {
       }
     }
   },
+  validations: {
+    form: {
+      incomeAmount: { required, minValue: v => v >= 0, decimals: decimals(2) },
+      pledgeAmount: { required, minValue: v => v >= 0, decimals: decimals(2) }
+    }
+  },
   computed: {
+    ...mapGetters([
+      'groupSettings',
+      'memberProfile'
+    ]),
+    currency () {
+      return currencies[this.groupSettings.mincomeCurrency]
+    },
     doesReceiveMonetary () {
       return !!this.fakeStore.receiving.monetary && !this.ephemeral.isEditingIncome
     },
     doesGiveMonetary () {
       return !!this.fakeStore.giving.monetary && !this.ephemeral.isEditingIncome
     }
   },
+  mounted () {
+    const profile = this.memberProfile(this.$store.state.loggedIn.username)
+    const incomeDetailsKey = profile && profile.groupProfile && profile.groupProfile.incomeDetailsKey
+    if (incomeDetailsKey) {
+      this.form.incomeDetailsKey = incomeDetailsKey
+      this.form[incomeDetailsKey] = profile.groupProfile[incomeDetailsKey]
+    }
+  },
   methods: {
+    async setPaymentInfo () {
+      if (this.$v.form.$invalid) {
+        alert(L('Invalid payment info'))
+        return
+      }
+      try {
+        const incomeDetailsKey = this.form.incomeDetailsKey
+        const groupProfileUpdate = await sbp('gi.contracts/group/groupProfileUpdate/create',
+          {
+            incomeDetailsKey,
+            [incomeDetailsKey]: this.form[incomeDetailsKey]
+          },
+          this.$store.state.currentGroupId
+        )
+        await sbp('backend/publishLogEntry', groupProfileUpdate)
+      } catch (e) {
+        console.error('setPaymentInfo', e)
+        alert(`Failed to update user's profile. Error: ${e.message}`)
```

Comment:
For some reason, each time I write on the input 50 or 100 (or any other number), it gives this error:  "Failed to update user's profile. Error: pledgeAmount has bad value: 50"